plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -1980,11 +1980,26 @@ function ( $collection_args ) {
 					);
 				}
 
+				$category_callback = function () use ( $collection_args ) {
+					return $collection_args['relatedBy']['categories'];
+				};
+
+				$tag_callback = function () use ( $collection_args ) {
+					return $collection_args['relatedBy']['tags'];
+				};
+
+				add_filter( 'woocommerce_product_related_posts_relate_by_category', $category_callback, PHP_INT_MAX );
+				add_filter( 'woocommerce_product_related_posts_relate_by_tag', $tag_callback, PHP_INT_MAX );
+
 				$related_products = wc_get_related_products(
 					$collection_args['relatedProductReference'],
```

Comment:
I know it's not in scope of this PR but would be good to keep collection name consistent across code:

```suggestion
					$collection_args['relatedProductsReference'],
```

That would be applied in multiple places.

Reply:
AFAIU This name is directly related to the `productReference` attribute name. Therefore, it's called `relatedProductReference`. But Happy to change the name if you think it's better to name it `relatedProductsReference`. I don't have strong feelings either way 😀

Reply:
Ah I see. 😅 Nvm!